private/protocol/jsonrpc/unmarshal_error.go

```diff
@@ -50,18 +70,33 @@ func (u *UnmarshalTypedError) UnmarshalError(
 	code := codeParts[len(codeParts)-1]
 	msg := jsonErr.Message
 
+	queryCodeParts := queryCodeParts(resp, u)
+
 	if fn, ok := u.exceptions[code]; ok {
 		// If exception code is know, use associated constructor to get a value
 		// for the exception that the JSON body can be unmarshaled into.
-		v := fn(respMeta)
+		var v error
+		queryErrFn, queryExceptionsFound := u.queryExceptions[code]
```

Comment:
instead of defining a separate `queryExceptions` map and keeping the `code` the same? why arent we changing the `code` instead when the query compatibility header is present?

Reply:
unmarshaller doesn't know about query-compatibility trait